### DISCUSSION TITLE
Unable to load script

### DISCUSSION THREAD

**Discussion by utkonos** [NationalSecurityAgency/ghidra#6449] [Q&A] [answered]
I am trying to isolate an error happening in all Java scripts. As an example, I am using the Skeleton script here:
https://github.com/astrelsky/vscode-ghidra-skeleton/blob/master/ghidra_scripts/ghidra_scripts/SkeletonScript.java

Here is the error:

```none
> Unable to load script: SkeletonScript.java
>   detail: The class could not be found. It must be the public class of the .java file: SkeletonScript
```

The code is copied and pasted into that java file named `SkeletonScript.java`.

**Comment by astrelsky** [comment]
> I am trying to isolate an error happening in all Java scripts. As an example, I am using the Skeleton script here: https://github.com/astrelsky/vscode-ghidra-skeleton/blob/master/ghidra_scripts/ghidra_scripts/SkeletonScript.java
> 
> Here is the error:
> 
> ```
> > Unable to load script: SkeletonScript.java
> >   detail: The class could not be found. It must be the public class of the .java file: SkeletonScript
> ```
> 
> The code is copied and pasted into that java file named `SkeletonScript.java`.

Fixing this will depend on your operating system. You'll need to go to the user ghidra folder where the application.log and preferences are stored. In that folder will be a folder named osgi. Delete the osgi folder and try again. I run into this problem all the time even with the scripts provided with Ghidra. It has become increasingly frustrating.

**Comment by utkonos** [comment]
@astrelsky I tried what you suggested, but the error is still there. No java-based scripts can run. Not even a hello world. I have also tried deleting the whole ~/.ghidra directory and still no change: nothing will run.

**Comment by utkonos** [comment, marked as answer]
I found the answer: Ghidra is not compatible with Java 22. Downgrading to 21.0.3 and the error no longer happens.

**Reply by ryanmkurtz** [reply]
Looks like we can upgrade some jars to get it working with Java 22.  I'll try to get that done for the next release.

**Comment by fenugrec** [comment]
Hmm downgrading java to 21.0.4 doesn't solve the issue here (ghidra 11.0.3 on linux)

**Reply by ryanmkurtz** [reply]
@fenugrec what version did you downgrade from?

**Reply by fenugrec** [reply]
it was 22.0.2

**Reply by fenugrec** [reply]
my bad, I had not deleted the `osgi` subfolder. So yes, downgrading to 21.0.4 appears to fix scripts
